pmd-cataloger/src/main/java/sfdc/sfdx/scanner/pmd/XmlFileFinder.java

```diff
@@ -34,39 +35,66 @@
 		}
 	}
 
+	/**
+	 * Represents a container for XML. This can either be a Jar file which may contain multiple XML files or an XML
+	 * file that contains itself.
+	 */
+	public static final class XmlContainer {
+		public final String filePath;
+		public final List<String> containedFilePaths;
+
+		public XmlContainer(String filePath) {
+			this(filePath, Collections.singletonList(filePath));
+		}
+
+		public XmlContainer(String filePath, List<String> containedFilePaths) {
+			this.filePath = filePath;
+			this.containedFilePaths = containedFilePaths;
+		}
+	}
 
 	/***
 	 * Find XML in a given path
 	 * @param pathString - could be a directory with classes and/or jar files, or a single jar file
 	 * @return a list of XML files found
 	 */
-	public List<String> findXmlFilesInPath(String pathString) {
+	public List<XmlContainer> findXmlFilesInPath(String pathString) {
 
 		final List<String> xmlFiles = new ArrayList<>();
 		final List<String> jarFiles = new ArrayList<>();
 
 		final Path path = Paths.get(pathString);
 
+		List<XmlContainer> xmlContainers = new ArrayList<>(); 
+
 		// Make sure that the path exists to begin with
 		if (!Files.exists(path)) {
 			throw new SfdxScannerException(EventKey.ERROR_INTERNAL_CLASSPATH_DOES_NOT_EXIST, path.getFileName().toString());
 		}
 
-
 		if (Files.isDirectory(path)) {
 			// Recursively walk through the directory and scout for XML/JAR files
 			xmlFiles.addAll(scoutForFiles(FileType.XML, path));
 			jarFiles.addAll(scoutForFiles(FileType.JAR, path));
 
-		} else if (Files.isRegularFile(path)
-			&& path.toString().endsWith(FileType.JAR.suffix)) { // Check if the path we have is a jar file
-
-			jarFiles.add(path.toString());
+		} else if (Files.isRegularFile(path)) {
+			if (path.toString().endsWith(FileType.JAR.suffix)) { // Check if the path we have is a jar file
+				jarFiles.add(path.toString());
+			} else if (path.toString().endsWith(FileType.XML.suffix)) {
```

Comment:
Add this to allow the user to specify an xml file that isn't contained in a jar. This is already supported by line 77 above if the xml file is contained in a directory that was passed to the method.